src/java/org/apache/cassandra/db/rows/UnfilteredRowIteratorWithLowerBound.java

```diff
@@ -181,9 +197,47 @@ public Row staticRow()
         return super.staticRow();
     }
 
-    private static <V> ClusteringBound<V> createInclusiveOpen(boolean isReversed, ClusteringPrefix<V> from)
+    private static <V> ClusteringBound<V> createArtificialLowerBound(boolean isReversed, ClusteringPrefix<V> from)
     {
-        return from.accessor().factory().inclusiveOpen(isReversed, from.getRawValues());
+        return !isReversed
+               ? from.accessor().factory().inclusiveOpen(false, from.getRawValues()).artificialLowerBound()
+               : from.accessor().factory().inclusiveOpen(true, from.getRawValues()).artificialUpperBound();
+    }
+
+    /**
+     * @return the lower bound stored on the index entry for this partition, if available.
+     */
+    private ClusteringBound<?> getKeyCacheLowerBound()
+    {
+        // NOTE: CASSANDRA-11206 removed the lookup against the key-cache as the IndexInfo objects are no longer
+        // in memory for not heap backed IndexInfo objects (so, these are on disk).
+        // CASSANDRA-11369 is there to fix this afterwards.
+
+        // Creating the iterator ensures that rowIndexEntry is loaded if available (partitions bigger than
+        // DatabaseDescriptor.column_index_size_in_kb)
+        maybeInit();
```

Comment:
The lower bound calculation should never initialize, this is the thing that it is meant to avoid.

Skip this part, along with both comments (the first is outdated), and go straight to `getCachedPosition`. If it's cached and useable, return the bound. If not, continue with `getMetadataLowerBound`.

Reply:
Right, if we initialized the iterator, we would lost the whole point of lower bound optimization